libs/stats/odc/stats/tasks.py

```diff
@@ -238,15 +263,19 @@ def msg_default(msg):
 
 
 class TaskReader:
-    def __init__(self,
-                 cache: Union[str, DatasetCache],
-                 product: Optional[OutputProduct] = None):
+    def __init__(
+        self, cache: Union[str, DatasetCache], product: Optional[OutputProduct] = None
+    ):
+        self.cache_path = None
         if isinstance(cache, str):
+            if "s3://" in cache:
+                self.cache_path = download_from_s3(cache)
```

Comment:
Should be the other way, you tell `download_from_s3` where to download data to.
Use https://docs.python.org/3.6/library/tempfile.html#tempfile.mkstemp to generate temp file name

Reply:
This has been refactored.